### PR TITLE
core: stdcm: fix linear allowance fallback and occupancy tolerance

### LINKED ISSUE #6285: core: stdcm: fix linear standard allowance used as fallback
* We actually need to look for conflicts and split allowance ranges, even when using linear allowance as fallback
* To do this, a tolerance needs to be added when identifying conflict times
* Requires #3482 to be fixed

### PR DESCRIPTION
**Pull request by eckter**
Fix https://github.com/osrd-project/osrd/issues/6285

* We needed to look for conflicts and split allowance ranges, even when using linear allowance as fallback
* To do this, a margin needs to be added when identifying conflicting times
* The PR also includes some smaller bugfixes, as the new version has a lower tolerance for errors
* Some tests need a higher tolerance to account for the conflict margin

### CODE REVIEW

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/ConstrainedEnvelopePartBuilder.java [hunk 1]
```diff
@@ -102,7 +104,7 @@ public boolean addStep(double position, double speed, double timeDelta) {
             position = intersection.position;
             speed = intersection.speed;
             if (position == lastPos) {
```
Comment (Erashin) [inline core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/ConstrainedEnvelopePartBuilder.java:106]: Wouldn't `arePositionsEqual` make sense here ?

File: core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMStandardAllowance.kt [hunk 2]
```diff
@@ -44,21 +44,23 @@ fun applyAllowance(
     comfort: Comfort?,
     blockAvailability: BlockAvailabilityInterface,
     departureTime: Double,
-    stops: List<TrainStop>
+    stops: List<TrainStop>,
+    isMareco: Boolean = true,
 ): Envelope {
     if (standardAllowance == null
         || standardAllowance.getAllowanceTime(envelope.totalTime, envelope.totalDistance) < 1e-5
```
Comment (Erashin) [inline core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMStandardAllowance.kt:51]: `areTimesEqual()` or use `TIME_EPSILON` ? (or whatever it's called)

**Review by Erashin — APPROVED** [review]

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/ConstrainedEnvelopePartBuilder.java (hunk 1 above)
Comment (eckter) [reply core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/ConstrainedEnvelopePartBuilder.java:106]: The envelopes mostly work with actual equalities when it comes to positions, if we add a tolerance here we would need other changes

File: core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMStandardAllowance.kt (hunk 2 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMStandardAllowance.kt:51]: `areTimesEqual` has been written on the other branch

It's not part of the diff, we can leave it as it is for now

**Pull request merged**